internal/scanner/walk.go

```diff
@@ -166,6 +169,16 @@ func (w *Walker) walkAndHashFiles(fchan chan protocol.FileInfo) filepath.WalkFun
 			return skip
 		}
 
+		if w.Pruner != nil {
```

Comment:
Isn't this a bit of an anti-pattern to check for nil receiver inside the function?